stellar/deploy-contract.js

```diff
@@ -101,35 +99,25 @@ async function processDeploy(options, config, chain) {
     await broadcast(operation, wallet, chain, 'Initialized contract', options);
 }
 
-async function processUpgrade(options, _, chain) {
-    const { contractName, contractId, privateKey, wasmPath, newWasmHash } = options;
+async function upgrade(options, _, chain) {
+    const { privateKey, wasmPath } = options;
     const { rpc, networkType } = chain;
     const networkPassphrase = getNetworkPassphrase(networkType);
+    const contractAddress = chain.contracts?.axelar_gateway.address;
 
-    if (!chain.contracts) {
-        chain.contracts = {};
+    if (!contractAddress) {
+        throw new Error('Stellar Gateway Address not found.');
     }
 
     const args = `--source ${privateKey} --rpc-url ${rpc} --network-passphrase "${networkPassphrase}"`;
 
-    let cmd;
+    let cmd = `${stellarCmd} contract install --wasm ${wasmPath} ${args}`;
+    const newWasmHash = execSync(cmd, { encoding: 'utf-8', stdio: 'pipe' }).trimEnd();
 
-    if (options.install) {
-        cmd = `${stellarCmd} contract install --wasm ${wasmPath} ${args}`;
-    } else {
-        cmd = `${stellarCmd} contract invoke --id ${contractId} ${args} -- upgrade --new_wasm_hash ${newWasmHash}`;
-    }
-
-    printInfo('Executing command', cmd);
-    printInfo('Deploying contract', contractName);
-
-    const result = execSync(cmd, { encoding: 'utf-8', stdio: 'pipe' }).trimEnd();
+    cmd = `${stellarCmd} contract invoke --id ${contractAddress} ${args} -- upgrade --new_wasm_hash ${newWasmHash}`;
+    execSync(cmd, { encoding: 'utf-8', stdio: 'pipe' }).trimEnd();
 
-    if (options.install) {
-        printInfo('Contract WASM hash', result);
-    } else {
-        printInfo('Upgraded contract successfully!');
-    }
+    printInfo('Contract upgraded successfully!', contractAddress);
 }
 
 async function mainProcessor(options, processor) {
```

Comment:
```suggestion
async function mainProcessor(options, processor, args) {
    const config = loadConfig(options.env);
    await processor(...args, options, config, getChainConfig(config, options.chainName));
    saveConfig(config, options.env);
}
```
might need to put `...args` at the end if JS doesn't allow it

Reply:
updated